client/src/model/apis/trainingsApi/trainingsApi.js

```diff
@@ -38,8 +38,12 @@ export const createTrainingsApi = ({ store, validator }) => {
     store.newTraining = null;
   };
 
-  const deleteNewExercise = () => {
-    store.newExercise = null;
+  const deleteNewExercise = (exId) => {
```

Comment:
seems you misunderstood the logic.
you shouldn't change function `deleteNewExercise`.
instead, you need to create another one `deleteExercise` by exId

Reply:
the same as it's with `newTraining` and `training`